### PR TITLE
[inductor] be able to enable max-autotune and cudagraphs independently

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98255

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by williamwen42 — APPROVED** [review]

File: torch/_inductor/__init__.py [hunk 1]
```diff
@@ -62,11 +62,16 @@ def list_mode_options(mode: str = None) -> Dict[str, Any]:
...
         "reduce-overhead": {
             "triton.cudagraphs": True,
         },
+        # enable max-autotune
+        "max-autotune-no-cudagraphs": {
+            "max_autotune": True,
+        },
+        # enable both cuda-graphs and max-autotune
         "max-autotune": {
-            "epilogue_fusion": True,
```
Comment (shunting314) [inline torch/_inductor/__init__.py:69]: remove this since it's True by default

**Pull request closed without merging**